spec/rspec/rails/matchers/action_cable/have_broadcasted_to_spec.rb

```diff
@@ -0,0 +1,220 @@
+        broadcast('stream', 'one')
+        broadcast('stream', 'two')
+      }.to have_broadcasted_to('stream').exactly(:twice)
+    end
+
+    it "passes with :thrice count" do
+      expect {
+        broadcast('stream', 'one')
+        broadcast('stream', 'two')
+        broadcast('stream', 'three')
+      }.to have_broadcasted_to('stream').exactly(:thrice)
+    end
+
+    it "passes with at_least count when sent messages are over limit" do
+      expect {
+        broadcast('stream', 'one')
+        broadcast('stream', 'two')
+      }.to have_broadcasted_to('stream').at_least(:once)
+    end
+
+    it "passes with at_most count when sent messages are under limit" do
+      expect {
+        broadcast('stream', 'hello')
+      }.to have_broadcasted_to('stream').at_most(:once)
+    end
+
+    it "generates failure message with at least hint" do
+      expect {
+        expect { }.to have_broadcasted_to('stream').at_least(:once)
+      }.to raise_error(/expected to broadcast at least 1 messages to stream, but broadcast 0/)
+    end
+
+    it "generates failure message with at most hint" do
+      expect {
+        expect {
+          broadcast('stream', 'hello')
+          broadcast('stream', 'hello')
+        }.to have_broadcasted_to('stream').at_most(:once)
+      }.to raise_error(/expected to broadcast at most 1 messages to stream, but broadcast 2/)
+    end
+
+    it "passes with provided data" do
+      expect {
+        broadcast('stream', id: 42, name: "David")
+      }.to have_broadcasted_to('stream').with(id: 42, name: "David")
+    end
+
+    it "passes with provided data matchers" do
+      expect {
+        broadcast('stream', id: 42, name: "David", message_id: 123)
+      }.to have_broadcasted_to('stream').with(a_hash_including(name: "David", id: 42))
+    end
+
+    it "generates failure message when data not match" do
+      expect {
+        expect {
+          broadcast('stream', id: 42, name: "David", message_id: 123)
+        }.to have_broadcasted_to('stream').with(a_hash_including(name: "John", id: 42))
+      }.to raise_error(/expected to broadcast exactly 1 messages to stream with a hash including/)
+    end
+
+    it "throws descriptive error when no test adapter set" do
+      require "action_cable/subscription_adapter/inline"
+      ActionCable.server.instance_variable_set(:@pubsub, ActionCable::SubscriptionAdapter::Inline)
+      expect {
+        expect { broadcast('stream', 'hello') }.to have_broadcasted_to('stream')
+      }.to raise_error("To use ActionCable matchers set `adapter: test` in your cable.yml")
+    end
+
+    it "fails with with block with incorrect data" do
+      expect {
+        expect {
+          broadcast('stream', "asdf")
+        }.to have_broadcasted_to('stream').with { |data|
+          expect(data).to eq("zxcv")
+        }
+      }.to raise_error { |e|
+        expect(e.message).to match(/expected: "zxcv"/)
+        expect(e.message).to match(/got: "asdf"/)
+      }
+    end
+
+
```

Comment:
Extra ws

Reply:
Done in https://github.com/rspec/rspec-rails/pull/2071/commits/86db9eb72fd42a24db3664cd5c4735b84dcfa59d